Detectors/GlobalTrackingWorkflow/tofworkflow/src/RecoWorkflowSpec.cxx

```diff
@@ -103,24 +91,17 @@ class TOFDPLRecoWorkflowTask
 
     // we do a copy of the input but we are looking for a way to avoid it (current problem in conversion form unique_ptr to *)
 
-    auto tracksRO = std::make_shared<std::vector<o2::dataformats::TrackTPCITS>>();
-    //  std::vector<o2::dataformats::TrackTPCITS> tracksRO;
-    *tracksRO.get() = std::move(*tracks);
-    // for (int i = 0; i < tracks->size(); i++) {
-    //   tracksRO.emplace_back(tracks->at(i));
-    // }
-    auto clustersRO = std::make_shared<std::vector<o2::tof::Cluster>>();
-    //std::vector<o2::tof::Cluster> clustersRO;
-    *clustersRO.get() = std::move(*clusters);
-    //  for (int i = 0; i < clusters->size(); i++) {
-    //    clustersRO.emplace_back(clusters->at(i));
-    // }
-
-    if (mUseMC)
-      mMatcher.initWorkflow(tracksRO.get(), clustersRO.get(), &toflab, itslab.get(), tpclab.get());
-    else
-      mMatcher.initWorkflow(tracksRO.get(), clustersRO.get(), nullptr, nullptr, nullptr);
+    o2::dataformats::MCTruthContainer<o2::MCCompLabel> toflab;
+    if (mUseMC) {
+      const auto toflabel = pc.inputs().get<o2::dataformats::MCTruthContainer<o2::MCCompLabel>*>("tofclusterlabel");
```

Comment:
these spans will be valid only in the local scope of ``{...}``,  define them outside, i.e.
````
gsl::span<o2::MCCompLabel> itslab;
gsl::span<o2::MCCompLabel> tpclab;
if (mUseMC) {
  ..
  itslab = pc.inputs().get<gsl::span<o2::MCCompLabel>>("itstracklabel");
  tpclab = pc.inputs().get<gsl::span<o2::MCCompLabel>>("tpctracklabel");
}
...
mMatcher.initWorkflow(tracksRO, clustersRO, toflab, itslab, tpclab);
````

Reply:
Changed accordingly to your suggestion (adding CONST: gsl::span<CONST o2::MCCompLabel> itslab;)
Thanks
Francesco

Reply:
Hi @noferini 
did you miss the comments above? I see you are still using the pointers on the span.

Reply:
Yes, I missed because for some reasons they were not diplayed. Now I covered them all.